sources/packages/backend/apps/queue-consumers/src/processors/schedulers/workflow/_tests_/assessment-workflow-enqueuer.scheduler.e2e-spec.ts

```diff
@@ -0,0 +1,237 @@
+      ).toBe(StudentAssessmentStatus.Queued);
+    });
+
+    it("Should queue the oldest pending reassessment when an active application has one completed original assessment and multiple submitted reassessments queued.", async () => {
+      // Arrange
+
+      // Application submitted with original assessment and completed assessment.
+      const application = await createDefaultApplication(
+        StudentAssessmentStatus.Completed,
+      );
+      // Oldest submitted assessment (the original one should be ignored because it is set as Completed.).
+      const oldestAssessment = createFakeStudentAssessment({
+        application,
+        auditUser,
+      });
+      oldestAssessment.triggerType = AssessmentTriggerType.OfferingChange;
+      oldestAssessment.createdAt = addDays(-1);
+      // Newest assessment that will be saved with current date and time.
+      const newestAssessment = createFakeStudentAssessment({
+        application,
+        auditUser,
+      });
+      newestAssessment.triggerType = AssessmentTriggerType.OfferingChange;
+      // Save all assessments.
+      await db.studentAssessment.save([oldestAssessment, newestAssessment]);
+      // Updates the current assessment to the newest one.
+      application.currentAssessment = newestAssessment;
+      await db.application.save(application);
+      // Queued job.
+      const job = createMock<Job<void>>();
+
+      // Act
+      await processor.enqueueAssessmentOperations(job);
+
+      // Assert
+      // Load application and related assessments for assertion.
+      const updatedApplication = await loadApplicationDataForAssertions(
+        application.id,
+      );
+      // Assert oldest submitted assessment was added to the queue.
+      const queueData = {
+        assessmentId: oldestAssessment.id,
+      } as StartAssessmentQueueInDTO;
+      expect(startApplicationAssessmentQueueMock.add).toBeCalledWith(queueData);
+      // currentProcessingAssessment must be updated with the oldest assessment
+      // and currentAssessment and currentProcessingAssessment must be different.
+      expect(updatedApplication.currentProcessingAssessment.id).toBe(
+        oldestAssessment.id,
+      );
+      expect(
+        updatedApplication.currentProcessingAssessment.studentAssessmentStatus,
+      ).toBe(StudentAssessmentStatus.Queued);
+      // Current assessment should not be updated.
+      expect(updatedApplication.currentAssessment.id).toBe(newestAssessment.id);
+    });
+
+    [
+      StudentAssessmentStatus.Queued,
+      StudentAssessmentStatus.InProgress,
+    ].forEach((status) => {
+      it(`Should not queue a '${StudentAssessmentStatus.Submitted}' assessment if the another assessment has its status as '${status}'`, async () => {
+        // Arrange
+
+        // Application submitted with original assessment.
+        const application = await createDefaultApplication(status);
+        const submittedAssessment = createFakeStudentAssessment({
```

Comment:
Should we not update `application.currentAssessment = submittedAssessment ` ?

Reply:
That operation happens inside the factory already used in the method `createDefaultApplication`.